tests/python/relax/frontend/test_onnx_frontend.py

```diff
@@ -0,0 +1,1378 @@
+
+        graph = helper.make_graph(
+            [fill_node],
+            "fill_test",
+            inputs,
+            initializer=[
+                helper.make_tensor(
+                    "input",
+                    TensorProto.INT64,
+                    [len(input_dim)],
+                    np.asarray(input_dim).astype("int64"),
+                )
+            ],
+            outputs=[
+                helper.make_tensor_value_info(
+                    "output", mapping.NP_TYPE_TO_TENSOR_TYPE[np.dtype(dtype)], input_dim
+                )
+            ],
+        )
+
+        model = helper.make_model(graph, producer_name="fill_test")
+        input_np = np.array(input_dim).astype("int64")
+        check_correctness(model, inputs={"input": input_np})
+
+    verify_constantofshape((2, 3, 4, 5), 10, "float32")
+    verify_constantofshape((3, 3), 0, "int32")
+    verify_constantofshape((1, 2, 3), -1, "float32")
+
+
+def test_slice():
+    def verify_slice(data_shape, output_shape, starts, ends, axes=None, steps=None):
+        if isinstance(starts, list):
+            starts = np.array(starts, "int64")
+        if isinstance(ends, list):
+            ends = np.array(ends, "int64")
+        if isinstance(axes, list):
+            axes = np.array(axes, "int64")
+        if isinstance(steps, list):
+            steps = np.array(steps, "int64")
+
+        slice_inputs = ["x", "starts", "ends"]
+        initializer = [
+            helper.make_tensor("starts", TensorProto.INT64, starts.shape, starts),
+            helper.make_tensor("ends", TensorProto.INT64, ends.shape, ends),
+        ]
+
+        if axes is not None:
+            initializer.append(helper.make_tensor("axes", TensorProto.INT64, axes.shape, axes))
+            slice_inputs.append("axes")
+        if steps is not None:
+            initializer.append(helper.make_tensor("steps", TensorProto.INT64, steps.shape, steps))
+            slice_inputs.append("steps")
+
+        slice_node = helper.make_node("Slice", inputs=slice_inputs, outputs=["y"])
+
+        graph = helper.make_graph(
+            [slice_node],
+            "slice_test",
+            inputs=[
+                helper.make_tensor_value_info("x", TensorProto.FLOAT, data_shape),
+            ],
+            outputs=[helper.make_tensor_value_info("y", TensorProto.FLOAT, output_shape)],
+            initializer=initializer,
+        )
+
+        model = helper.make_model(graph, producer_name="slice_test")
+        check_correctness(model)
+
+    # Test with all parameters set.
+    verify_slice([20, 10, 5], [3, 10, 5], starts=[0, 0], ends=[3, 10], axes=[0, 1], steps=[1, 1])
+    # Test with default axes and steps.
+    verify_slice([20, 10, 5], [3, 10, 5], starts=[0, 0], ends=[3, 10])
+    # Test with negative steps.
+    verify_slice(
+        [20, 10, 5],
+        [19, 3, 2],
+        starts=[20, 10, 4],
+        ends=[0, 0, 1],
+        steps=[-1, -3, -2],
+        axes=[0, 1, 2],
+    )
```

Comment:
Missing permutation: Including steps but not axes or including axes but not steps

Reply:
Test added

Reply:
I also discovered an issue that needs to be addressed at some point regarding optional inputs